services/addFilter.js

```diff
@@ -0,0 +1,20 @@
+export default function addFilter(array, filter) {
+  const date = new Date();
+  const time = date.getHours();
+  const month = date.getMonth() + 1;
+
+  const filterSpecies = el => !filter?.species || el.species === filter.species;
```

Comment:
```suggestion
  const filterSpecies = el => el.species === !filter.species ? el.species : filter.species;
```

This could be another way to handle the filter. It really depends on which is easier to read for you I guess, imo my suggestion is easier to read.